pkg/execution/state/redis_state/queue.go

```diff
@@ -1958,6 +1965,7 @@ func (q *queue) Scavenge(ctx context.Context) (int, error) {
 		Limit(0, 100).
 		Build()
 
+	// NOTE: Received keys can be legacy (workflow IDs or system/internal queue names) or new (full Redis keys)
 	pKeys, err := q.u.unshardedRc.Do(ctx, cmd).AsStrSlice()
```

Comment:
This still works even when an old executor receives new keys as we store the partition item with the new key and ignore non-default partitions